### DISCUSSION TITLE
Pipelines and Cloud Risks

### DISCUSSION THREAD

**Discussion by tdiprima** [open-webui/pipelines#315] [General]
[One article](https://ikasten.io/2024/06/03/getting-started-with-openwebui-pipelines/) noted, "Pipelines is agnostic to the UI client, as long as the client supports the OpenAI API," which typically refers to cloud-based services. I've noticed some filters and pipelines make internet requests (e.g., Wikipedia). Does this mean that any filter or pipeline could potentially reach out to the cloud because of OpenAI's involvement?

**Comment by tdiprima** [comment]
Is it true that the pipeline organizes and tweaks your input, but the final response generation&mdash;where the LLM interprets the prompt and creates an answer&mdash;comes from OpenAI's cloud-based model, not from a model running on your computer?

**Comment by milafrerichs** [comment]
no, the reference to OpenAI API just refers to the specification of an OpenAPI client (https://github.com/openai/openai-openapi?tab=readme-ov-file). If you're using Ollama it does have an OpenAPI endpoint you can add to your UI.
